Gifski/CustomButton.swift

```diff
@@ -225,6 +225,22 @@ open class CustomButton: NSButton {
 		needsDisplay = true
 	}
 
+	public typealias ColorGenerationHandler = () -> NSColor
+
+	private var colorGenerators = [KeyPath<CustomButton, NSColor>: ColorGenerationHandler]()
+
+	public func setColorGenerator(for keyPath: KeyPath<CustomButton, NSColor>, handler: @escaping ColorGenerationHandler) {
+		colorGenerators[keyPath] = handler
+	}
```

Comment:
What do you think about making this setter/getter instead?

Then you could just use it like this:


```swift
// Add
$0.colorGenerator(for: \.backgroundColor) = {
	NSColor.appTheme.with(alpha: 0.1)
}

// Remove
$0.colorGenerator(for: \.backgroundColor) = nil
```

Reply:
I've tried getting close to the syntax you came up with, but I'm not sure it's possible without adding something like `$0.colorGenerator(for: \.backgroundColor).value`. 🤔

I've created a PR that just makes the handler optional, allowing for `$0.setColorGenerator(for: \.backgroundColor, nil)`.

Reply:
Yeah, you're right. Too bad that doesn't work though. Would have been nice to have this. I should propose something like that on the Swift Forums.

Could make computed properties support arguments:

```swift
var unicorn(for: Magic): Int {
    get { … }
    set { … }
}
```

Used as:

```swift
.unicorn(for: .rainbow) = 2
```


Thoughts on this?